presto-parquet/src/main/java/com/facebook/presto/parquet/cache/MetadataReader.java

```diff
@@ -70,6 +70,12 @@
 
     public static ParquetFileMetadata readFooter(ParquetDataSource parquetDataSource, long fileSize)
             throws IOException
+    {
+        return readFooter(parquetDataSource, fileSize, System.currentTimeMillis());
```

Comment:
why do we still need the readFooter(ParquetDataSource parquetDataSource, long fileSize)? should we always use the readFooter with a modification time?

Reply:
Using currentTime might make the cached footer invalidated, is this you really intend to do？

Reply:
> why do we still need the readFooter(ParquetDataSource parquetDataSource, long fileSize)? should we always use the readFooter with a modification time?

Thank you for your advice, @beinan . Because Delta and Iceberg don't use parquet meta cache, the cache is only used in `ParquetPageSourceFactory` for `HivePageSourceProvider`:  
![image](https://user-images.githubusercontent.com/26461591/159847169-b29c27f9-7b9c-4719-aeb6-a4f2c3e80fd4.png)  
Iceberg and Delta also use the  `MetadataReader` API to get metadata from HDFS or other storages(without using the cache), and the invalidation function for now only supports hive file (i.e. `HivePageSourceProvider`), i need the `fileModifiedTime` in `HiveFileContext`, which is not exists in `DeltaPageSourceProvider` and `IcebergPageSourceProvider`. So i just use this API to be compatible with the originnal logic of them.
![image](https://user-images.githubusercontent.com/26461591/159845211-85741667-6b25-4cf6-8680-c7ad049bf00e.png)  
Using currentTime is just a kind of API compatibility for Iceberg and Delta, because they don't use `CachingParquetMetadataSource`, that means they don't use parquet meta cache.

Reply:
Make sense, I'm convinced to keep the old readFooter.  But if it's just for API compatibility, is it possible to use 0L instead of System.currentTime?  I'm thinking if we could define a constant `long MODIFICATION_TIME_NOT_SET = 0L`, what do you think?

Reply:
> Make sense, I'm convinced to keep the old readFooter. But if it's just for API compatibility, is it possible to use 0L instead of System.currentTime? I'm thinking if we could define a constant `long MODIFICATION_TIME_NOT_SET = 0L`, what do you think?

Good idea~Sure i will add that constant, and push again.